ui/raidboss/data/06-ew/raid/p2n.ts

```diff
@@ -0,0 +1,178 @@
+import NetRegexes from '../../../../../resources/netregexes';
+import Outputs from '../../../../../resources/outputs';
+import { callOverlayHandler } from '../../../../../resources/overlay_plugin_api';
+import { Responses } from '../../../../../resources/responses';
+import ZoneId from '../../../../../resources/zone_id';
+import { RaidbossData } from '../../../../../types/data';
+import { PluginCombatantState } from '../../../../../types/event';
+import { TriggerSet } from '../../../../../types/trigger';
+
+export interface Data extends RaidbossData {
+  bodyActor?: PluginCombatantState;
+}
+
+const triggerSet: TriggerSet<Data> = {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  zoneId: ZoneId.AsphodelosTheSecondCircle,
+  timelineFile: 'p2n.txt',
+  triggers: [
+    {
+      id: 'P2N Murky Depths',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '680F', source: 'Hippokampos', capture: false }),
+      response: Responses.aoe(),
+    },
+    {
+      id: 'P2N Doubled Impact',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '680E', source: 'Hippokampos' }),
+      response: Responses.sharedTankBuster(),
+    },
+    {
+      id: 'P2N Spoken Cataract',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['67F8', '67F7', '67F9'], source: 'Hippokampos' }),
+      delaySeconds: 1,
+      promise: async (data) => {
+        const callData = await callOverlayHandler({
+          call: 'getCombatants',
+        });
+        if (!callData || !callData.combatants || !callData.combatants.length) {
+          console.error('SpokenCataract: failed to get combatants: ${JSON.stringify(callData)}');
+          return;
+        }
+        // This is the real hippo, according to hp.
+        const hippos = callData.combatants.filter((c) => c.BNpcID === 13721);
+        if (hippos.length !== 1) {
+          console.error('SpokenCataract: There is not exactly one Hippo?!?: ${JSON.stringify(hippos)}');
+          data.bodyActor = undefined;
+          return;
+        }
+        data.bodyActor = hippos[0];
+      },
+      alertText: (data, matches, output) => {
+        if (!data.bodyActor) {
+          console.error('SpokenCataract: No boss actor found. Did the promise fail?');
+          return;
+        }
+        const heading = data.bodyActor?.Heading;
```

Comment:
This is a minor comment, but do you need the `?` here if you've already checked `data.bodyActor` above? If typescript is complaining, can you store this in its own variable, e.g. `const bodyActor = data.bodyActor` and then say `if (!bodyActor) etc` so that you don't need this? I think logically it's not possible for `data.bodyActor` to be undefined here, but it makes it harder to read if it looks like it might.

Reply:
Fixed in a8562bbece06d2fd3a33b1cbac9473feb52ee86b